src/networking.c

```diff
@@ -1028,7 +1028,7 @@ void addReplyAttributeLen(client *c, long length) {
 
 void addReplyPushLen(client *c, long length) {
     serverAssert(c->resp >= 3);
-    serverAssertWithInfo(c, NULL, c->flags & CLIENT_PUSHING);
+    serverAssertWithInfo(c, NULL, c->flags & CLIENT_PUSHING || c != server.current_client);
```

Comment:
Can we document this, I still can't really wrap my head around why this is here.

Reply:
Right, I'll document it (unless we change the API so we don't need it).

When we push to a client that's not the current client, there's no risk the push will arrive inside multi-exec. Thus, it's safe to allow this.

From the module API, I don't set CLIENT_PUSHING, because it's hard to know when when to clear the flag, i.e. where the push reply ends.